src/lib/config/index.ts

```diff
@@ -245,10 +219,21 @@ export class ShellConfig {
     }
   }
 
+  validate = () => {
+    if (this.endpoint === undefined) {
+      // No `~/.fauna-shell` was found, and no `--secret` was passed.
+      throw new Error(
+        "No endpoint or secret set. Set an endpoint in ~/.fauna-shell, .fauna-project, or pass --endpoint"
+      );
+    }
+  };
```

Comment:
we do almost all of the validation in the constructor for this class and then just do the endpoint in the validate method.  It seems that we are doing it this way because some commands don't require an endpoint.  Which ones are those?
Given that it seems we are saying that this class is technically 'valid' without an endpoint.  With that, I think I'd rather have the code paths that need it ensure there is an endpoint without this method on this class.

Reply:
yup, this should be renamed and refactored a bit to move more validation into here.